src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -544,24 +583,27 @@ object IEnumerator.Current
 
             public bool MoveNext()
             {
-                if (_version == _builder._version)
+                ReadOnlyCollectionBuilder<T> localBuilder = _builder;
```

Comment:
This is specific to this issue, not general problem of "up for grabs".
Even if something is "approved" or "up for grabs", there can always be unanticipated problems and side effects in the implementation -- the developers are supposed to catch these during coding or at least in code review and discuss.

For this particular case of breaking changes, we should wait for @VSadov @OmarTawfik @bartdesmet @JonHanna blessing on what kind of breaking changes are acceptable in this library.